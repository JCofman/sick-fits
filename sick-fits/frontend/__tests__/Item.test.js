import ItemComponent from "../components/Item";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";
const fakeItem = {
  id: "ABCD",
  title: "A cool Item",
  price: 5000,
  description: "This is a nice item",
  image: "cool.jpg",
  largeImage: "largeCool.jpg"
};

describe("<Item>", () => {
  it("renders and displays propery", () => {
    const wrapper = shallow(<ItemComponent item={fakeItem} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
