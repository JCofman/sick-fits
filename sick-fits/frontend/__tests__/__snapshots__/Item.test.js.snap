// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item> renders and displays propery 1`] = `
<ItemStyles__Item>
  <img
    alt="A cool Item"
    src="cool.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "ABCD",
          },
        }
      }
    >
      <a>
         
        A cool Item
      </a>
    </Link>
  </Title>
   
  <PriceTag>
    $50
  </PriceTag>
  <p>
    This is a nice item
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "/update",
          "query": Object {
            "id": "ABCD",
          },
        }
      }
    >
      <a>
        Edit ✏️
      </a>
    </Link>
    <AddToCart
      id="ABCD"
    />
    <DeleteItem
      id="ABCD"
    >
      Delete this Item
    </DeleteItem>
  </div>
</ItemStyles__Item>
`;
